sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala

```diff
@@ -682,25 +684,25 @@ class XmlTokenizer(
         return false
       }
       val c = cOrEOF.toChar
-      if (c == commentEnd(i)) {
-        if (i >= commentEnd.length - 1) {
-          // Found comment close.
+      if (c == end(i)) {
+        i += 1
+        if (i >= end.length) {
```

Comment:
Please add a test with two scenarios:
- CDATA ends at the end of the file,
- CDATA never ends. 
The later will be invalid XML. Goal is to make sure the parser doesn't crash and still returns other valid records.

Add the same two tests for comments as well.

Reply:
Thanks for the review and test cases!